src/util/auth.rs

```diff
@@ -24,28 +24,49 @@ pub fn check_auth(
 /// Checks if user is authorized by checking if the given Bearer Token matches the given hash
 pub fn check_auth_header(
     authorization: Authorization<Bearer>,
-    hash: &PasswordHashString,
+    hashes: &Vec<PasswordHashString>,
 ) -> Result<(), (StatusCode, String)> {
-    return check_auth_key(authorization.token().as_bytes(), hash);
+    return check_auth_key(authorization.token().as_bytes(), hashes);
 }
 
 /// Checks if user is authorized by checking if a given query parameter matches the given hash
 pub fn check_auth_query(
     authorization: &String,
-    hash: &PasswordHashString,
+    hash: &Vec<PasswordHashString>,
 ) -> Result<(), (StatusCode, String)> {
     return check_auth_key(authorization.as_bytes(), hash);
 }
 
 /// Checks authorization by checking if a (raw) key matches a given hash  
 /// Returns 401 (UNAUTHORIZED) with appropriate message if they do not match
-pub fn check_auth_key(key: &[u8], hash: &PasswordHashString) -> Result<(), (StatusCode, String)> {
-    return match (Argon2::default()).verify_password(key, &hash.password_hash()) {
-        Err(err) => {
-            // TODO: Might want to distinguish between invalid password (4xx) and internal errors (e.g. cryptographic ones)
-            log::error!("Error during authentication: {}", err);
-            return Err((StatusCode::UNAUTHORIZED, "Invalid token!".to_string()));
+pub fn check_auth_key(
+    key: &[u8],
+    hashes: &Vec<PasswordHashString>,
+) -> Result<(), (StatusCode, String)> {
+    let mut error: Option<password_hash::errors::Error> = None;
+    let mut err_hash: Option<&PasswordHashString> = None;
+
+    for hash in hashes {
+        match (Argon2::default()).verify_password(key, &hash.password_hash()) {
+            Ok(_) => return Ok(()),
+            Err(err) => {
+                error = Some(err);
+                err_hash = Some(hash);
```

Comment:
Das sorgt doch einfach dafür, dass in den Variablen jeweils der letzte gefailte Hash drinsteht, oder nicht?

Reply:
Korrekt, am richtigsten wäre es jeden gefailten Hash mitzunotieren. Dachte mir zumindest einen Failure anzuzeigen ist besser als keiner und wenn man anfängt alle zu Speichern muss man Vec nutzen oÄ, was ich schon für bisschen much halte für error logging

Reply:
Bin bei dir, der change von dir mit check auf den return value ist mMn die schönere Option